packages/runtime/container-runtime/src/test/compatUtils.spec.ts

```diff
@@ -0,0 +1,154 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { FlushMode } from "@fluidframework/runtime-definitions/internal";
+
+import {
+	getConfigsForCompatMode,
+	type IContainerRuntimeOptionsVersionDependent,
+	type SemanticVersion,
+} from "../compatUtils.js";
+import {
+	disabledCompressionConfig,
+	enabledCompressionConfig,
+} from "../compressionDefinitions.js";
+
+describe("compatUtils", () => {
+	describe("getConfigsForCompatMode", () => {
+		it('returns correct configs compatibilityMode "pre-3.0-default"', () => {
+			const result = getConfigsForCompatMode("pre-3.0-default", testConfigMap);
+			// We should return the hardcoded default configs for pre-3.0-default, even if using a different config map
+			assert.deepEqual(result, {
+				gcOptions: {},
+				flushMode: FlushMode.TurnBased,
+				compressionOptions: enabledCompressionConfig,
+				enableRuntimeIdCompressor: undefined as unknown as "on" | "delayed",
+				enableGroupedBatching: true,
+				explicitSchemaControl: false,
+			});
+		});
+		const testConfigMap: {
+			[K in keyof IContainerRuntimeOptionsVersionDependent]: {
+				[version: SemanticVersion]: IContainerRuntimeOptionsVersionDependent[K];
+			};
+		} = {
+			enableGroupedBatching: {
+				"1.0.0": false,
+				"2.3.4": true,
+			},
+			compressionOptions: {
+				"1.0.0": disabledCompressionConfig,
+				"2.1.0": enabledCompressionConfig,
+			},
+			enableRuntimeIdCompressor: {
```

Comment:
i want to see more test of different numbers of versions, 0,1,2,3,10, both in order and out of order